lib/config/index.js

```diff
@@ -113,6 +121,20 @@ if (config.gitlab && config.gitlab.version !== 'v4' && config.gitlab.version !==
 // Only update i18n files in development setups
 config.updateI18nFiles = (env === Environment.development)
 
+// Verify that sourceURL is correct
+if (env === Environment.production) {
+  request.get(config.sourceURL)
```

Comment:
Just to have this on record: I'm not a fan of pinging home or anywhere on server startup. Not from a technological standpoint but more so not from a surveillance standpoint. I don't need to tell Github/Microsoft and the US Government where my server is located every time CodiMD comes up.

Reply:
That's right. That's why people are free to set the sourceURL. The reason to verify the source URL is that we have a plenty of forks out there and a lot of them have modifications. Most of them still don't change any data about the repository location or similar.

This PR adds a link to the entry page which points to the source code. It makes no sense to add this link when the link is wrong (which it would be, for all these mentioned forks). So it would mislead the user to a source code that isn't actually the source code of the instance they are using.

So I'm open for alternative solutions for that problem, but the idea is to catch forks, that are unaware that they have to change something to get this link right. It doesn't protect from people who want to fool users intentionally, but at least unintentional fooling should be prevented.

As mentioned in Matrix. One of the alternative thoughts I had was zipping the CodiMD directory. But that again is dangerous because someone may keeps backups of the config.json, places private keys next to the server, … and this way would cause a high risk for deployments. So the alternative is to link to the correct source and the probably worst UX is when you end up in a dead end. And in this case it has [legal implications](https://github.com/hackmdio/codimd/blob/32af96aa376576d060e16a2c0ff7ed5719d24e91/LICENSE#L542-L551).

As AGPL states you not only have to make your users aware that they CAN recieve a copy of the source code (which we do with the link), you also have to provide it in a way that it can be recieved from a server at no charge. Granted, this could also mean you send out an email to them, but I think this is quite uncommon these days and probably up to legal interpretation. Still I think misleading/wrong links are worse than pinging GitHub. (Which is also avoidable when you set the link to an own page that provides the source code.)